accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/aem/AemCryptoSupport.java

```diff
@@ -9,8 +9,15 @@
 package biz.netcentric.cq.tools.actool.aem;
 
 /** OSGi service to encapulate AEM crypto support (to make it optional) */
-public interface AemCryptoSupport {
+public abstract class AemCryptoSupport {
 
-    public String unprotect(String password);
+    public abstract String unprotect(String password);
+
+    /**
+     * Copied from CryptoImpl, need it separately to be able to check if decryption is necessary before CryptoSupport is loaded

```

Comment:
Looking at this I first thought it wouldn't work because AemCryptoSupport does not become available if the dependency CryptoSupport is not there. Now I see it is a static method and hence it should work. I think it's confusing though to make the interface an abstract class just to allow for this static method. I would rather create a class `AemCryptoSupportHelper` with a sole static method `isProtected()`, then it's at least clear what is happening and `AemCryptoSupport` can remain an interface

Reply:
I don't get the concern here: Coming up with a separate class just to place one static method in it feel too heavy. What is you concern of converting the interface to an abstract class?
(Since Java 8 even interfaces are allowed to have static methods, but we are yet bound to Java 7)

Reply:
I think it's better for maintainability to have this separated (an extra class does not cost anything and clearly separates the "decoupling of AEM crypto support" aspect from the utility method that is always needed (leaving AemCryptoSupport as interface makes that easier)

Reply:
For me those two belong together. The utilty method needs to mimic exactly what the CryptoSupportImpl is doing (otherwise it is less useful)

> an extra class does not cost anything
It makes it harder to see where it belongs to